### PR TITLE
Add doc examples for `Vec::{as_slice,as_mut_slice}`.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libcollections/vec.rs [hunk 1]
```diff
@@ -541,6 +541,14 @@ impl<T> Vec<T> {
     /// Extracts a slice containing the entire vector.
     ///
     /// Equivalent to `&s[..]`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io::{self, Write};
+    /// let buffer = vec![1, 2, 3, 5, 8];
+    /// let _ = io::sink().write(buffer.as_slice());
```
Comment (steveklabnik) [inline src/libcollections/vec.rs]: i would prefer `unwrap` to `let _`, the former at least doesn't fail silently.

Comment (frewsxcv) [inline src/libcollections/vec.rs]: Latest force push switched to `unwrap`.

**Pull request merged**